src/EVEMon.Common/Notifications/SkillCompletionNotificationEventArgs.cs

```diff
@@ -47,7 +47,9 @@ public override void Append(NotificationEventArgs other)
             List<QueuedSkill> skills = ((SkillCompletionNotificationEventArgs)other).Skills.ToList();
             foreach (QueuedSkill skill in skills.Where(skill => !Skills.Contains(skill)))
             {
-                Skills.Add(skill);
+                if (Skills.ToList().FindIndex(f => f.SkillName == skill.SkillName) == -1) {
```

Comment:
It doesn't work, every timer tick the same skills get added over and over again. If you leave client open long enough itshows hundreds of the same skill in the list

Reply:
Did this actually fix it? I am still looking at this and it appears that the actual bug is in SkillQueue#UpdateOnTimerTick

Reply:
It fixed the part where skills get duplicated making the list grow. It didn't fix the bit where notifications keep firing every timer tick even if you reviewed it and deleted.

Reply:
So `.Where(skill => !Skills.Contains(skill))` did not work because `QueuedSkill` has the default equality comparator, so different instances with the same name would be unequal. However, this patch fails if you train a skill up two levels back to back, as it will suppress the notification of the second level being finished. I fixed this upstream (not yet pushed) by defining a sensible `Equals` and `GetHashCode` method in `QueuedSkill`.

How do I take half of this PR? The labels change is good.

Reply:
I'll do separate PR for labels

Reply:
I'll do separate PR for labels